autoblocks/_impl/tracer.py

```diff
@@ -44,10 +46,27 @@ def __init__(
         self._trace_id: Optional[str] = trace_id
         self._properties: Dict = properties or {}
 
-        self._client = httpx.Client(
-            headers={"Authorization": f"Bearer {ingestion_key}"},
-            timeout=timeout.total_seconds(),
-        )
+        ingestion_key = ingestion_key or AutoblocksEnvVar.INGESTION_KEY.get()
+        if not ingestion_key:
+            raise ValueError(
+                f"You must provide an ingestion_key or set the {AutoblocksEnvVar.INGESTION_KEY} environment variable."
+            )
+
+        if not AutoblocksTracer._client:
+            AutoblocksTracer._client = httpx.Client(
+                headers={"Authorization": f"Bearer {ingestion_key}"},
+                timeout=timeout.total_seconds(),
+            )
+            return
+
+        if AutoblocksTracer._client.headers.get("authorization") != f"Bearer {ingestion_key}":
+            raise ValueError(
+                "You must use the same configuration for all AutoblocksTracer instances in your application."
+            )
+        if AutoblocksTracer._client.timeout != httpx.Timeout(timeout.total_seconds()):
+            raise ValueError(
+                "You must use the same configuration for all AutoblocksTracer instances in your application."
+            )
```

Comment:
this is a little awkward but probably the easiest way forward in the SDK's current state to allow creating new instances of the tracer on every request w/o creating a new client every time. likely in future versions we will want to decouple the client and the tracer